etl/extractor/platform_impl.go

```diff
@@ -188,7 +188,7 @@ func (e *PlatformExtractor) retrieveRecords(ctx context.Context, pu *exporter.Fu
 				if strings.Contains(err.Error(), "trying to get a non-finalized pulse data") ||
 					strings.Contains(err.Error(), "pulse not found") {
 					time.Sleep(time.Duration(e.continuousPulseRetrievingHalfPulseSeconds) * time.Second)
-					go e.retrievePulses(ctx, int64(pu.PrevPulseNumber), int64(pu.PulseNumber-1)) // goroutine to split the stack
+					go e.retrievePulses(ctx, int64(pu.PrevPulseNumber), int64(pu.PulseNumber)) // goroutine to split the stack
```

Comment:
why did you change it?

Reply:
`retrievePulses` should load only pulse `pu.PulseNumber` here, and we dont need to do `-1` to have this behavior: then we get pu.PulseNumber as result of `GetNextFinalizedPulse`, we stop anyway (condition to stop - `>=`)
I think `-1` is unnecessary complexity

Reply:
let's merge and watch logs. I think i can show the problem if it appears.